Cargo.toml

```diff
@@ -12,7 +12,7 @@ libc = ">=0.2.39"
 log = ">=0.4.6"
 vhost = { version = "0.4", features = ["vhost-user-slave"] }
 virtio-bindings = "0.1"
-virtio-queue = "0.4"
+virtio-queue = { git = "https://github.com/andreeaflorescu/vm-virtio.git", branch = "refactor_queue" }
```

Comment:
@andreeaflorescu could you please update this so that it relies on latest upstream branch:
```
virtio-queue = { git = "https://github.com/rust-vmm/vm-virtio.git", branch = "main" }
```

Reply:
Yes, I will do that today, I am also adding the error handling and changing the tests. Will do a push later today.

Reply:
Ok, looks like I did most changes on Friday already and there wasn't much more to do. I pushed a new revision.

Reply:
Thanks! I've been able to update the Cloud Hypervisor's port. Here's the complete PR https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4290
Waiting for `virtio-queue` to cut a new release and this PR to be merged on `vhost-user-backend` (followed by a release as well). In the meantime I'm gonna perform some performance testing to be sure this new implementation doesn't introduce any regression.

Reply:
Thanks Sebastien. I will need to do another small update to the mock interface before the release to handle errors properly. This should not have any impact on the production code.

Reply:
@sboeuf do you have an update on the performance runs? Is there any impact resulting from the patch series? I just published v0.5.0, and updated this PR as well.

Reply:
Yes I've run the performance test suite we have and everything looks good! So we can move forward!